### PR TITLE
Multiple vectors per document support in nearest neighbor search.

### PR DESCRIPTION
**Pull request by geirst**
Also introduce Vespa's vector concept (tensors) and simplify the examples.

@jobergum please review (but do not merge).

After first review I will follow-up with a commit that updates all the other reference docs.

### CODE REVIEW

File: en/nearest-neighbor-search.md [hunk 1]
```diff
@@ -5,36 +5,82 @@ redirect_from:
...
-Many real-world applications require query-time constrained vector search. 
-For example, real-time recommender systems using [vector search for candidate retrieval](tutorials/news-5-recommendation.html) 
-need to filter recommendations by hard constraints like regional availability or age group suitability. 
-Likewise, search systems using vector search need to support filtering.
-For example, typical e-commerce search interfaces allow users to navigate and filter search results using result facets. 
 
-## Using Vespa's nearest neighbor search
+## Vectors in Vespa
+In Vespa a vector is represented by a [tensor](tensor-user-guide.html) with one indexed dimension.
+Example [tensor type](reference/tensor.html#tensor-type-spec) representing a float vector with 384 dimensions:
```
Comment (jobergum) [inline en/nearest-neighbor-search.md:20]: Maybe we should link to the cell precision types supported by tensors and nearest neighbor search

**Review by jobergum — APPROVED** [review]

File: en/nearest-neighbor-search.md (hunk 1 above)
Comment (geirst) [reply en/nearest-neighbor-search.md:20]: The linked doc (reference/tensor.html#tensor-type-spec) lists the four cell precision types: double, float, bfloat16, int8, so this should be ok.

**Comment by geirst** [comment]
@jobergum please review the last 2 commits.

**Comment by jobergum** [comment]
Ship it!

**Pull request merged**